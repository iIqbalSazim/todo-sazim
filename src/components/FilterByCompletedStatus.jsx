import { Tabs } from "@mantine/core";
import { COLORS } from "../constants/constant";

const FilterByCompletedStatus = ({ setFilter }) => {
  return (
    <Tabs
      orientation="vertical"
      variant="pills"
      color={COLORS.BUTTON}
      defaultValue="all"
      onTabChange={(value) => setFilter(value)}
      h={"180px"}
    >
      <Tabs.List position="apart">
        <Tabs.Tab value="all">All</Tabs.Tab>
        <Tabs.Tab value="active">Active</Tabs.Tab>
        <Tabs.Tab value="completed">Completed</Tabs.Tab>
      </Tabs.List>
    </Tabs>
  );
};

export default FilterByCompletedStatus;
