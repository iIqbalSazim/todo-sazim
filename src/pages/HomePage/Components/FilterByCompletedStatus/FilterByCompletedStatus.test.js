import { render, screen } from "@testing-library/react";
import FilterByCompletedStatus from "./FilterByCompletedStatus";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

const mockSetFilter = jest.fn();

describe("FilterByCompletedStatus", () => {
  test("renders correctly", () => {
    const { container } = render(
      <FilterByCompletedStatus setFilter={mockSetFilter} />
    );

    const allTab = screen.getByRole("tab", {
      name: /all/i,
    });
    const activeTab = screen.getByRole("tab", {
      name: /active/i,
    });
    const completedTab = screen.getByRole("tab", {
      name: /completed/i,
    });

    expect(allTab).toBeInTheDocument();
    expect(activeTab).toBeInTheDocument();
    expect(completedTab).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test("calls setFilter with value 'all' when clicked on All tab", async () => {
    const { container } = render(
      <FilterByCompletedStatus setFilter={mockSetFilter} />
    );

    const allTab = screen.getByRole("tab", {
      name: /all/i,
    });

    await userEvent.click(allTab);

    expect(mockSetFilter).toBeCalledWith("all");

    expect(container).toMatchSnapshot();
  });

  test("calls setFilter with value 'active' when clicked on Active tab", async () => {
    const { container } = render(
      <FilterByCompletedStatus setFilter={mockSetFilter} />
    );

    const activeTab = screen.getByRole("tab", {
      name: /active/i,
    });

    await userEvent.click(activeTab);

    expect(mockSetFilter).toBeCalledWith("active");

    expect(container).toMatchSnapshot();
  });

  test("calls setFilter with value 'completed' when clicked on Completed tab", async () => {
    const { container } = render(
      <FilterByCompletedStatus setFilter={mockSetFilter} />
    );

    const completedTab = screen.getByRole("tab", {
      name: /completed/i,
    });

    await userEvent.click(completedTab);

    expect(mockSetFilter).toBeCalledWith("completed");

    expect(container).toMatchSnapshot();
  });
});
