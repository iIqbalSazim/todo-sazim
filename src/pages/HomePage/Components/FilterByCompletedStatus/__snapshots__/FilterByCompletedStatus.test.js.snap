// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByCompletedStatus calls setFilter with value 'active' when clicked on Active tab 1`] = `
<div>
  <div
    class="mantine-Tabs-root mantine-o79pz9"
  >
    <div
      aria-orientation="vertical"
      class="mantine-Tabs-tabsList mantine-1dphvmd"
      role="tablist"
    >
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-ayflxk12m-tab-all"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          All
        </span>
      </button>
      <button
        aria-selected="true"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        data-active="true"
        id="mantine-ayflxk12m-tab-active"
        role="tab"
        tabindex="0"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Active
        </span>
      </button>
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-ayflxk12m-tab-completed"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Completed
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FilterByCompletedStatus calls setFilter with value 'all' when clicked on All tab 1`] = `
<div>
  <div
    class="mantine-Tabs-root mantine-o79pz9"
  >
    <div
      aria-orientation="vertical"
      class="mantine-Tabs-tabsList mantine-1dphvmd"
      role="tablist"
    >
      <button
        aria-selected="true"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        data-active="true"
        id="mantine-1iomv7rt4-tab-all"
        role="tab"
        tabindex="0"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          All
        </span>
      </button>
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-1iomv7rt4-tab-active"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Active
        </span>
      </button>
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-1iomv7rt4-tab-completed"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Completed
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FilterByCompletedStatus calls setFilter with value 'completed' when clicked on Completed tab 1`] = `
<div>
  <div
    class="mantine-Tabs-root mantine-o79pz9"
  >
    <div
      aria-orientation="vertical"
      class="mantine-Tabs-tabsList mantine-1dphvmd"
      role="tablist"
    >
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-sedor4wcv-tab-all"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          All
        </span>
      </button>
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-sedor4wcv-tab-active"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Active
        </span>
      </button>
      <button
        aria-selected="true"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        data-active="true"
        id="mantine-sedor4wcv-tab-completed"
        role="tab"
        tabindex="0"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Completed
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FilterByCompletedStatus renders correctly 1`] = `
<div>
  <div
    class="mantine-Tabs-root mantine-o79pz9"
  >
    <div
      aria-orientation="vertical"
      class="mantine-Tabs-tabsList mantine-1dphvmd"
      role="tablist"
    >
      <button
        aria-selected="true"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        data-active="true"
        id="mantine-n2z4uw42f-tab-all"
        role="tab"
        tabindex="0"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          All
        </span>
      </button>
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-n2z4uw42f-tab-active"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Active
        </span>
      </button>
      <button
        aria-selected="false"
        class="mantine-UnstyledButton-root mantine-Tabs-tab mantine-1l1luj0"
        id="mantine-n2z4uw42f-tab-completed"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <span
          class="mantine-480qv8 mantine-Tabs-tabLabel"
        >
          Completed
        </span>
      </button>
    </div>
  </div>
</div>
`;
